src/Services/MarketTradingDataService.cs

```diff
@@ -116,6 +164,69 @@ public MarketTradingDataService(ICandlesHistoryServiceProvider candlesHistorySer
             return candles;
         }
 
+        private async
+            Task<(IReadOnlyDictionary<string, Candle> spotCandle,
+                IReadOnlyDictionary<string, Candle> mtCandle)> GetLastCandlesAsync()
+        {
+            var spotCandlesTask = GetLastCandlesAsync(MarketType.Spot);
+            var mtCandlesTask = GetLastCandlesAsync(MarketType.Mt);
+
+            var spotCandlesData = await spotCandlesTask;
+            var mtCandlesData = await mtCandlesTask;
+
+            return (spotCandle: spotCandlesData, mtCandle: mtCandlesData);
+        }
+
+        private async Task<(Candle spotCandle, Candle mtCandle)> GetLastCandleAsync(
+            string assetPair)
+        {
+            var spotCandlesTask = GetLastCandleAsync(MarketType.Spot, assetPair);
+            var mtCandlesTask = GetLastCandleAsync(MarketType.Mt, assetPair);
+
+            var spotCandlesData = await spotCandlesTask;
+            var mtCandlesData = await mtCandlesTask;
+
+            return (spotCandle: spotCandlesData, mtCandle: mtCandlesData);
+        }
+
+        private async Task<IReadOnlyDictionary<string, Candle>> GetLastCandlesAsync(MarketType market)
+        {
+            var assetPairs = await GetAvailableAssetPairsAsync(market);
+            var candlesService = _candlesHistoryServiceProvider.Get(market);
+            var to = DateTime.UtcNow.AddHours(1); // exclusive
+            var from = DateTime.UtcNow; // inclusive
```

Comment:
Usage of different UtcNow call for ```for``` to and ```from``` could be a bug cause to. If these two calls will be occured in different hours, then ```to``` and ```from``` will point at the same hour and no candles will be returned. Replace with:
```
from = DateTime.UtcNow;
to = from.AddHours(1);
```